### PR TITLE
FS-928 Optimize bom cmd according to thread discussion.

### PR DESCRIPTION
**Pull request by Alva8756**
This PR optimized the bom cmd format according to discussion in slack thread https://equinix.slack.com/archives/C056JN319PG/p1695317935291199

example:
```
mctl % ./mctl create bom --from-xlsx-file ~/Downloads/test-alva1.xlsx
 &{0 0 0 0 0 {0x1400061a1f0 <nil> <nil> <nil> <nil>} resource created   <nil> <nil>}
mctl % ./mctl get bom --aoc-mac B5                                   
[
  {
    "serial_num": "A5",
    "aoc_mac_address": "B5,B6",
    "bmc_mac_address": "bmc1,bmc2",
    "num_defi_pmi": "3CECEFCF0087",
    "num_def_pwd": "FIWUZMZZGQ",
    "metro": ""
  }
]
mctl % ./mctl get bom --bmc-mac bmc1                                 
[
  {
    "serial_num": "A5",
    "aoc_mac_address": "B5,B6",
    "bmc_mac_address": "bmc1,bmc2",
    "num_defi_pmi": "3CECEFCF0087",
    "num_def_pwd": "FIWUZMZZGQ",
    "metro": ""
  }
]
```
error example:
```
mctl % ./mctl get bom                  
 --aoc-mac and --bmc-mac not set

mctl % ./mctl get bom --aoc-mac B5 --bmc-mac b2
Error: if any flags in the group [aoc-mac bmc-mac] are set none of the others can be; [aoc-mac bmc-mac] were all set
```
Tested both locally and in sandbox.

### CODE REVIEW

File: cmd/get/bom.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+var getBomInfoByMacAddress = &cobra.Command{
+	Use:   "bom",
+	Short: "Get bom object by AOC or BMC Addr",
+	Run: func(cmd *cobra.Command, args []string) {
+		if !flagsGetBomByMacAddress.hasAOCMacAddr() && !flagsGetBomByMacAddress.hasBMCMacAddr() {
+			log.Fatalf("--aoc-mac and --bmc-mac not set")
+		}
+
+		if flagsGetBomByMacAddress.hasAOCMacAddr() && flagsGetBomByMacAddress.hasBMCMacAddr() {
+			log.Fatalf("--aoc-mac and --bmc-mac should not set simutansly")
```
Comment (DoctorVin) [inline cmd/get/bom.go:38]: Spelling on `simultaneously`. Also, defer this to Cobra: https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagsMutuallyExclusive

**Review by DoctorVin — CHANGES_REQUESTED** [review]
Small change requested. Looks fine otherwise.

File: cmd/get/bom.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ var getBomInfoByMacAddress = &cobra.Command{
 		}
 
 		if flagsGetBomByMacAddress.hasAOCMacAddr() && flagsGetBomByMacAddress.hasBMCMacAddr() {
```
Comment (DoctorVin) [inline cmd/get/bom.go:37]: With cobra marking these as mutually exclusive, I don't think we'll ever get here. I think the library will terminate on an error.

**Review by DoctorVin — CHANGES_REQUESTED** [review]
I know it's kind of a hassle, but if you could remove the test that will never come back true that would be best.

**Comment by Alva8756** [comment]
> I know it's kind of a hassle, but if you could remove the test that will never come back true that would be best.

Thanks Vince,
I am lost here. Can you please help explain more about "remove the test"? I didn't add test to this PR.
Do you suggest I paste logs that's sending invalid requests and getting failures?

**Comment by DoctorVin** [comment]
> > I know it's kind of a hassle, but if you could remove the test that will never come back true that would be best.
> 
> Thanks Vince, I am lost here. Can you please help explain more about "remove the test"? I didn't add test to this PR. Do you suggest I paste logs that's sending invalid requests and getting failures?

I meant “remove the if statement”. You got it.

**Pull request merged**